src/components/admin/settings-tab.js

```diff
@@ -46,55 +44,52 @@ function SettingsTab({ onDirtyStateChange }) {
   }, [settingsApiClient, t]);
 
   useEffect(() => {
-    onDirtyStateChange(!!dirtyKeys.length || !!invalidKeys.length);
-  }, [dirtyKeys, invalidKeys, onDirtyStateChange]);
+    onDirtyStateChange(isDirty);
+  }, [isDirty, onDirtyStateChange]);
 
-  const handleChange = useCallback((key, value, isValid) => {
+  const handleChange = useCallback((key, value) => {
+    setIsDirty(true);
     setSettings(prev => ({ ...prev, [key]: value }));
-    setDirtyKeys(prev => ensureIsIncluded(prev, key));
-    setInvalidKeys(prev => isValid ? ensureIsExcluded(prev, key) : ensureIsIncluded(prev, key));
-  }, [setSettings, setDirtyKeys, setInvalidKeys]);
+  }, [setSettings, setIsDirty]);
 
   const handleHomepageInfoChange = useCallback(event => {
-    handleChange('homepageInfo', event.target.value, true);
+    handleChange('homepageInfo', event.target.value);
   }, [handleChange]);
 
-  const handleConsentTextChange = useCallback((value, { isValid }) => {
-    handleChange('consentText', value, isValid);
+  const handleConsentTextChange = useCallback(value => {
+    handleChange('consentText', value);
   }, [handleChange]);
 
   const handleTemplateDocumentChange = useCallback(documentId => {
-    handleChange('templateDocument', { documentId }, true);
+    handleChange('templateDocument', { documentId });
   }, [handleChange]);
 
-  const handleHelpPageChange = useCallback((value, { isValid }) => {
-    handleChange('helpPage', value, isValid);
+  const handleHelpPageChange = useCallback(value => {
+    handleChange('helpPage', value);
```

Comment:
Great that you simplified and cleaned up all this useless validation here!

Reply:
It's cleanup week, so I thought ...